src/golang/cmd/server/handler/preview.go

```diff
@@ -75,6 +81,59 @@ func (*PreviewHandler) Name() string {
 	return "Preview"
 }
 
+func checkError(w http.ResponseWriter, err error) {
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		return
+	}
+}
+
+func (*PreviewHandler) SendResponse(w http.ResponseWriter, response interface{}) {
+
+	structResponse := response.(previewResponse)
+
+	responseNoArtifacts := previewResponseNoArtifacts{
```

Comment:
Let's change this to `responseMetadata` and also include the artifact metadata such as the serialization type and the artifact type into this struct, so we don't have to write these metadata into the multi-part form file.